src/yaml/from_yaml.cr

```diff
@@ -22,39 +29,110 @@ end
 
 def Nil.new(pull : YAML::PullParser)
   location = pull.location
-  value = pull.read_scalar
-  if value.empty?
+  case value = pull.read_plain_scalar
+  when "", "~", "null", "Null", "NULL"
     nil
   else
-    raise YAML::ParseException.new("Expected nil, not #{value}", *location)
+    raise YAML::ParseException.new("Expected null, not '#{value}'", *location)
   end
 end
 
 def Bool.new(pull : YAML::PullParser)
-  pull.read_scalar == "true"
+  location = pull.location
+  case value = pull.read_plain_scalar
+  when "y", "Y", "yes", "Yes", "YES", "true", "True", "TRUE", "on", "On", "ON"
+    true
+  when "n", "N", "no", "No", "NO", "false", "False", "FALSE", "off", "Off", "OFF"
+    false
+  else
+    raise YAML::ParseException.new("Expected boolean, not '#{value}'", *location)
+  end
 end
 
-{% for type in %w(Int8 Int16 Int32 Int64 UInt8 UInt16 UInt32 UInt64) %}
-  def {{type.id}}.new(pull : YAML::PullParser)
+{% for size in %w(8 16 32 64) %}
```

Comment:
I'd rather keep the types here and read the bytesize from the type string. This is more expressiv and more concise.